syne_tune/optimizer/schedulers/single_fidelity_scheduler.py

```diff
@@ -52,15 +52,18 @@ class SingleFidelityScheduler(TrialScheduler):
     def __init__(
         self,
         config_space: Dict[str, Any],
-        metric: str,
+        metric: List[str] | str,
```

Comment:
Could we perhaps inherit rather? (Handling either list or single metrics was causing a lot of errors in our previous version)

Reply:
I also thought about it. What do you think about allowing only a list of metrics to `SingleFidelityScheduler`, i.e making it multi-objective only, and inherit a `SingleObjectiveScheduler` that only expect a single metric? The idea here is that single-objective optimization is a special case of multi-objective optimization.

Reply:
That can work also, if we can avoid to use `metric` to refer to a list of metrics it would be much better.

Reply:
I am wondering if we should do the same thing for `BaseSearcher`?